src/main/java/org/example/parampammusic/service/AlbumService.java

```diff
@@ -0,0 +1,60 @@
+package org.example.parampammusic.service;
+
+import org.example.parampammusic.entity.Album;
+import org.example.parampammusic.repository.AlbumRepository;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class AlbumService {
+
+    private final AlbumRepository albumRepository;
+
+    public AlbumService(AlbumRepository albumRepository) {
+        this.albumRepository = albumRepository;
+    }
+
+    public List<Album> getAllAlbums() {
+        return albumRepository.findAll();
+    }
+
+    public Album getAlbumById(int albumId) {
+        return albumRepository.findById(albumId)
+                .orElseThrow(() -> new IllegalArgumentException("Album with id " + albumId + " not found."));
+    }
+
+    public void addAlbum(Album album) {
+        albumRepository.save(album);
+    }
+
+    public Album updateAlbum(Album album, Authentication authentication) {
+        Optional<Album> existingAlbum = albumRepository.findById(album.getId());
+
+        if (existingAlbum.isPresent()) {
+            if (isAdmin(authentication)) {
```

Comment:
This validation should be moved to validator and execute in the controller level